clippy_lints/src/instant_elapsed.rs

```diff
@@ -0,0 +1,90 @@
+    /// ```rust
+    /// use std::time::Instant;
+    /// let prev_instant = Instant::now();
+    /// let duration = Instant::now() - prev_instant;
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// use std::time::Instant;
+    /// let prev_instant = Instant::now();
+    /// let duration = prev_instant.elapsed();
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub INSTANT_ELAPSED,
+    pedantic,
+    "catch substraction between Instant::new() and previous timestamp "
+}
+
+#[derive(Default)]
+pub struct InstantElapsed {
+    variable_name: String,
+    local_id: u32,
+}
+
+impl_lint_pass!(InstantElapsed => [INSTANT_ELAPSED]);
+
+impl LateLintPass<'_> for InstantElapsed {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &'_ Expr<'_>) {
+        if let ExprKind::Block(block, _) = expr.kind {
+            for (_, stmt) in block.stmts.iter().enumerate() {
+                if let StmtKind::Local(local) = stmt.kind
+                    && let PatKind::Binding(BindingAnnotation::Unannotated, local_hir_id, ident, None) = local.pat.kind
+                    && let Some(expr_block) = local.init
+                    && check_instant_now_call(expr_block) {
+                    self.variable_name = ident.to_string();
+                    self.local_id = local_hir_id.local_id.as_u32();
+                }
+
+                if let StmtKind::Local(local) = stmt.kind
+                    && let Some(expr_block) = local.init
+                    && let ExprKind::Binary(Spanned {node: BinOpKind::Sub, ..}, lhs, rhs) = &expr_block.kind
+                    && check_instant_now_call(lhs)
+                    && let ExprKind::Path(qpath) = &rhs.kind
+                    && let QPath::Resolved(None, rhs_path) = qpath
+                    && let rustc_hir::def::Res::Local(expr_res) = rhs_path.res
+                    && expr_res.local_id.as_u32() == self.local_id {
+
+                    let local_var_name = if let Some(Node::Pat(res_pat)) = cx.tcx.hir().find(expr_res) && let PatKind::Binding(_, _, ident, _)  = res_pat.kind { ident.name.to_string() } else { "".to_string() }; 
+
+                    span_lint_and_sugg(
+                        cx,
+                        INSTANT_ELAPSED,
+                        expr_block.span,
+                        "substract between two instant",
+                        "try",
+                        format!("{}.elapsed()", local_var_name),
+                        Applicability::MaybeIncorrect,
+                    );
+                }
+            }
+        }
+    }
+}
+
+fn check_instant_now_call(expr_block: &'_ Expr<'_>) -> bool {
+    if let ExprKind::Call(func, []) = expr_block.kind
+    && let ExprKind::Path(QPath::TypeRelative(ty, name)) = func.kind
+    && name.ident.as_str() == "now"
+    && let TyKind::Path(ty_path) = &ty.kind
+    && let QPath::Resolved(None, path) = ty_path
+    && path.segments[0].ident.name.as_str() == "Instant" {
+        true
+    } else {
+        false
+    }
```

Comment:
Here you can add the `Instant` path to `clippy_utils::paths`, then use `clippy_utils::match_def_path` on the id from `clippy_utils::path_def_id(cx, ty_path)`. This will ensure we're getting the `std` version of `Instant` and not any other type named that

Reply:
Thanks